tket/src/Placement/Placement.cpp

```diff
@@ -154,6 +154,45 @@ std::vector<qubit_mapping_t> Placement::get_all_placement_maps(
   return {get_placement_map(circ_)};
 }
 
+qubit_mapping_t NaivePlacement::get_placement_map(const Circuit &circ_) const {
+  return get_all_placement_maps(circ_).at(0);
+}
+
+std::vector<qubit_mapping_t> NaivePlacement::get_all_placement_maps(
+    const Circuit &circ_) const {
+  qubit_mapping_t placement;
+  qubit_vector_t to_place;
+  std::vector<Node> placed;
+
+  // Find which/if any qubits need placing
+  for (const Qubit &q : circ_.all_qubits()) {
+    Node n(q);
+    if (!this->arc_.node_exists(n)) {
+      to_place.push_back(n);
+    }
+    if (this->arc_.node_exists(n)) {
+      placed.push_back(n);
+      // if already placed, make sure qubit retains placement
+      placement.insert({n, n});
+    }
+  }
```

Comment:
Maybe set a variable `unsigned n_placed = to_place.size();` here and use it in the places below where we call `to_place.size()`?

Reply:
done